benchmark/testsystems/bookkeepers.py

```diff
@@ -220,30 +196,49 @@ def sample_v_given_x(self, x):
         self.simulation.context.applyVelocityConstraints(self.constraint_tolerance)
         return get_velocities(self.simulation)
 
-    def accumulate_shadow_work(self, x_0, v_0, n_steps, store_potential_energy=False):
+    def accumulate_shadow_work(self, x_0, v_0, n_steps, box_vectors=None, store_potential_energy=False, store_W_shad_trace=False):
         """Run the integrator for n_steps and return the change in energy - the heat."""
-        get_energy = lambda: get_total_energy(self.simulation)
+        get_energy = lambda: get_total_energy(self.simulation).value_in_unit(W_unit)
         get_potential = lambda: get_potential_energy(self.simulation).value_in_unit(W_unit)
         get_heat = lambda: self.simulation.integrator.getGlobalVariableByName("heat")
 
-        set_positions(self.simulation, x_0)
-        set_velocities(self.simulation, v_0)
-
         result = {}
 
+        try:
+            set_positions(self.simulation, x_0, box_vectors=box_vectors)
+            set_velocities(self.simulation, v_0)
+        except:
+            print("Error setting positions or velocities!")
+            return result
+
         # Apply position and velocity constraints.
         self.simulation.context.applyConstraints(self.constraint_tolerance)
         self.simulation.context.applyVelocityConstraints(self.constraint_tolerance)
 
         E_0 = get_energy()
         Q_0 = get_heat()
 
-        if store_potential_energy:
-            potential_energies = [get_potential()]
+        if store_potential_energy or store_W_shad_trace:
+            if store_potential_energy:
+                potential_energies = [get_potential()]
+            if store_W_shad_trace:
+                total_energies = [E_0]
+                heats = [Q_0]
+                W_shad_trace = []
             for _ in range(n_steps):
                 self.simulation.step(1)
-                potential_energies.append(get_potential())
-            result["potential_energies"] = potential_energies
+                if store_potential_energy:
+                    potential_energies.append(get_potential())
+                if store_W_shad_trace:
+                    total_energies.append(get_energy())
+                    heats.append(get_heat())
+                    DeltaE = total_energies[-1] - total_energies[-2]
+                    DeltaQ = heats[-1] - heats[-2]
+                    W_shad_trace.append(DeltaE - DeltaQ)
```

Comment:
This is an efficiency / ease-of-implementation issue. For MTS schemes, asking the integrator to monitor the `shadow_work` property can result in many potential energy evaluations per timestep, so it can be more efficient to compute the total energy change outside the integrator and subtract the integrator-accumulated `heat`.

Re: correctness, I believe there's a test in `test_shadow_work.py` that confirms this scheme is consistent with the integrator `shadow_work` property.

A later optimization would be to do this bookkeeping more efficiently _inside_ the integrator, but I think it's unnecessary for now.